### PR TITLE
Python client added features

### PR DESCRIPTION
**Pull request by sutartmelson**
Modified downloadFile function to allow downloading to a file-like object rather than just downloading to a local directory. 

Added listFile function that is similar to listFolder and listItem. It simply lists all the files in a given item.

Added optional limit parameter to the aforementioned list functions.

### CODE REVIEW

File: clients/python/girder_client/__init__.py [hunk 1]
```diff
@@ -288,6 +288,21 @@ def listResource(self, path, params):
         """
         return self.get(path, params)
 
+    def listFiles(self, itemId, limit=None):
```
Comment (zachmullen) [inline clients/python/girder_client/__init__.py]: Let's call this `listFile` so it agrees with the naming of the other methods.

**Comment by zachmullen** [comment]
One last thing, could you add a test for your new method? [This file](https://github.com/girder/girder/blob/master/tests/cases/py_client/lib_test.py) contains our testing of the python client library.

**Comment by sutartmelson** [comment]
I've written tests for all of the code I wrote/altered.

**Comment by zachmullen** [comment]
LGTM

**Pull request merged**